llvm/include/llvm/ProfileData/SampleProfReader.h

```diff
@@ -380,6 +380,22 @@ class SampleProfileReader {
     return sampleprof_error::success;
   }
 
+  /// Read sample profiles for the given functions. Currently it's only used for
+  /// extended binary format to load the profiles on-demand.
+  std::error_code read(const DenseSet<StringRef> &FuncsToUse) {
+    if (std::error_code EC = read(FuncsToUse, Profiles))
+      return EC;
+    return sampleprof_error::success;
+  };
+
+  /// Read sample profiles for the given functions and write them to the given
+  /// profile map. Currently it's only used for extended binary format to load
+  /// the profiles on-demand.
+  virtual std::error_code read(const DenseSet<StringRef> &FuncsToUse,
```

Comment:
As mentioned in the other PR, would be good to make API simpler -- always load into the same profile map owned by reader, without needing a profile map parameter.

Reply:
Make it private function? Otherwise we now have a public API without a check for repeated load for a function. Also there is not need to expose this as API?